src/zimbraweb/emlparsing.py

```diff
@@ -0,0 +1,91 @@
+from typing import Tuple
+import re
+import base64
+
+from email.parser import Parser
+from zimbraweb import ZimbraUser, WebkitAttachment  # for parsing eml
+
+
+def eml_parsing(user: ZimbraUser, eml: str) -> Tuple[bytes, str]:
+    """Generate a payload from any eml
+
+        Parameters:
+            user: ZimbraUser Class
+            eml: str
+
+        Returns:
+            bytes: The WebkitFormBoundary Payload
+            str: The boundary used in the payload
+    """
+
+    parser = Parser()
+    parsed = parser.parsestr(eml)
+
+    if type(parsed.get_payload()) == str:
+        return user.generate_webkit_payload(parsed['To'], parsed['Subject'], parsed.get_payload())
+
+    elif type(parsed.get_payload()) == list:
+        dict_mail = {}
+        dict_mail['to'] = parsed['To']
+        dict_mail['subject'] = parsed['Subject']
+        dict_mail['attachments'] = []
+        return user.generate_webkit_payload(**multipart_eml_parsing(user, parsed.get_payload(), dict_mail))
+
+    else:
+        raise TypeError("Multipart Payload in Plain Parser. Please use multipart_eml_parsing")
+
+
+def plain_eml_parsing(user: ZimbraUser, eml: str) -> Tuple[bytes, str]:
+    # right now this function is useless, since its so short and basically needs to be done for eml_parsing function anyway
+    """Generate a payload from plaintext eml
+
+        Parameters:
+            user: ZimbraUser Class
+            eml: str
+
+        Returns:
+            bytes: The WebkitFormBoundary Payload
+            str: The boundary used in the payload
+    """
+
+    parser = Parser()
+    parsed = parser.parsestr(eml)
+
+    # this if is not strictly necessary, but prevents from calling the plain function for multipart messages
+    if type(parsed.get_payload()) == str:
+        return user.generate_webkit_payload(parsed['To'], parsed['Subject'], parsed.get_payload())
+    else:
+        raise TypeError("Multipart Payload in Plain Parser. Please use multipart_eml_parsing")
```

Comment:
As I said in the comment, I recognise that this function is currently not used. I will remove it until it may become necessary to pull it out of `parse_eml`